### DISCUSSION TITLE
Side-by-side metrics collection with Prometheus and OTel

### DISCUSSION THREAD

**Discussion by andreminelli** [open-telemetry/opentelemetry-dotnet#4135] [Q&A]
Is there a guide (or tip) on how to allow that (legacy) Prometheus metrics and (new) OTel metrics could be 'merged' in a single Prometheus-format output? This would allow a much simpler migration path, until all legacy Prometheus instrumentation could be changed to OTel API.

Prometheus library has a public method that exports its data to an arbitrary Stream, but OTel doesn't have such support (`PrometheusExporter` class is internal at this time).

**Comment by andreminelli** [comment]
I just discovered that `prometheus-net` also exports metrics from `System.Diagnostics.Metrics`, so this already works.

**Comment by martincostello** [comment]
By coincidence, I also have a blog post about this pending: https://github.com/open-telemetry/opentelemetry.io/pull/11281
